app/components/views/LNPage/ChannelsTab/hooks.js

```diff
@@ -1,31 +1,137 @@
+  const autopilotEnabled = useSelector(sel.lnAutopilotEnabled);
+  const [nodeErrorMsg, setNodeErrorMsg] = useState(null);
+  const [nodeShowSuccess, setNodeShowSuccess] = useState(false);
 
   const {
-    walletBalances,
-    channelBalances,
     channels,
     pendingChannels,
     closedChannels,
     isMainNet,
     openChannel,
-    closeChannel
+    closeChannel,
+    describeGraph
   } = useLNPage();
 
-  const onNodeChanged = (e) => {
-    const _canOpen = e.target.value && localAmtAtoms > 0;
-    setNode((e.target.value || "").trim());
+  const recentNodes = useMemo(
+    () =>
+      [
+        ...new Set(
+          [...pendingChannels, ...channels, ...closedChannels].map(
+            (c) => c.remotePubkey
+          )
+        )
+      ].map((pubKey) => {
+        const nodeDetails = describeGraph?.nodeList?.find(
+          (node) => node.pubKey === pubKey
+        );
+        return { pubKey: pubKey, alias: nodeDetails?.alias || pubKey };
+      }),
+    [channels, pendingChannels, closedChannels, describeGraph]
+  );
+
+  const hideSearchBt = describeGraph?.nodeList?.length <= 1 ?? true;
+
+  const filteredChannels = useMemo(() => {
+    return [...pendingChannels, ...channels, ...closedChannels]
+      .filter(
+        (channel) =>
+          !channelFilter ||
+          !channelFilter.type ||
+          channelFilter.type === "all" ||
+          channelFilter.type === channel.status
+      )
+      .filter(
+        (channel) =>
+          !channelFilter ||
+          !channelFilter.search ||
+          channel.channelPoint
+            .toLowerCase()
+            .indexOf(channelFilter.search.toLowerCase()) !== -1
+      );
+  }, [channels, pendingChannels, closedChannels, channelFilter]);
+
+  const recentlyOpenedChannel = useMemo(
+    () =>
+      recentlyOpenedChannelChannelPoint
+        ? [...channels, ...pendingChannels].find(
+            (c) => c.channelPoint === recentlyOpenedChannelChannelPoint
+          )
+        : null,
+    [channels, pendingChannels, recentlyOpenedChannelChannelPoint]
+  );
+
+  const onNodeChanged = (value) => {
+    const node = (value || "").trim();
+    const isNodeValid = validateNode(node);
+    const _canOpen = node && localAmtAtoms > 0 && isNodeValid;
+    setNode(node);
     setCanOpen(_canOpen);
   };
 
+  const validateNode = (node) => {
+    setNodeErrorMsg(null);
+    setNodeShowSuccess(false);
+    const split = node.split("@");
+    if (split.length > 2) {
+      setNodeErrorMsg(intl.formatMessage(messages.invalidNodeFormat));
+      return false;
+    }
+
+    let nodePubKey;
+    if (split.length == 2) {
+      nodePubKey = split[0];
+      const addressSplit = split[1].split(":");
+      if (addressSplit.length > 2) {
+        setNodeErrorMsg(intl.formatMessage(messages.invalidNodeAddressFormat));
+        return false;
+      }
+    } else {
+      nodePubKey = node;
+    }
+    if (!secp256k1.publicKeyVerify(Buffer.from(nodePubKey, "hex"))) {
```

Comment:
Weirdly, seems this is missing to validate some obviously wrong ids. Not sure wether the secp256k1 lib here is not processing correctly or the `Buffer.from` call is silencing errors.

For example (`038fde001dbe4d6286ab168cfd1e9711ad0cbb8fc4e3c2312f2b42063b72af8e71` is a valid and existing ID on testnet):

```
038fde001dbe4d6286ab168cfd1e9711ad0cbb8fc4e3c2312f2b42063b72af8e7x  (last digit changed to invalid hex char `x`)
038fde001dbe4d6286ab168cfd1e9711ad0cbb8fc4e3c2312f2b42063b72af8e71xxx (added invalid hex chars)
03ce0b14fb842b1ba549fdd675c98075f12e9c510f8ef52bd021a9a1f4809d3b4c (invalid pubkey as tested [here](https://github.com/decred/dcrd/blob/7474441bbe947c718db98522e37701a9b27023e3/dcrec/secp256k1/pubkey_test.go#L106-L109))

Reply:
@matheusd It seems the `publicKeyVerify` function doesn't examine (properly) whether a point is on the curve. I've added an additional check using directly the elliptic library. Also, add hex validation.
Thank you for your review.